source/docs/upstream_filters.md

```diff
@@ -0,0 +1,32 @@
+# Converting a downstream filter to be a dual filter
+
+Before coverting a filter to be an upstream filter you should do some basic
+functionality analysis. Make sure
+
+  * The filter does not use any downstream-only functionality, such as clearing
+    cached routes, or performing internal redirects.
+  * Either the filter does not sendLocalReply, or you thoroughly document how
+    sendLocalReply code paths will play with hedging / retries (cut off the
+    hedge attempt, and local-reply failures won't trigger retries)
+
+Once you've done this, you're ready to convert. An example converted filter is the Envoy
+[Buffer]
+(https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/buffer/config.cc)
+[Filter](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/buffer/config.h)
+
```

Comment:
I think the markdown might need some slight update https://github.com/alyssawilk/envoy/blob/docs/source/docs/upstream_filters.md
Rendered markdown page looks like below 👇

---
Once you've done this, you're ready to convert. An example converted filter is the Envoy
[Buffer]
(https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/buffer/config.cc)
[Filter](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/buffer/config.h)